persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java

```diff
@@ -29,8 +30,9 @@ public JpaStoreConfigurationBuilder entityClass(Class<?> entityClass) {
       return self();
    }
 
+   @Deprecated
    public JpaStoreConfigurationBuilder batchSize(long batchSize) {
-      attributes.attribute(BATCH_SIZE).set(batchSize);
+      attributes.attribute(MAX_BATCH_SIZE).set((int) batchSize);
```

Comment:
Not normally, but some people may pass in `Long.MAX_VALUE`, which would convert to -1 :(

To be honest a lot of our config code is missing validations. For example we wouldn't want a number 0 or less passed in as well.

However in regards to `> Integer.MAX_VALUE` I would just use `Integer.MAX_VALUE` as the batch size.